src/pudl/output/eia860.py

```diff
@@ -136,11 +138,168 @@ def plants_eia860(pudl_engine, start_date=None, end_date=None):
+      as their most consistent BA code because we know ``SWPP`` accumulated many smaller
+      balancing authorities
+    * use the backfilled code for plants that have a differnt backfilled and most
+      consistent code, but the most consistent code is has a consistency rate (how
+      frequent the most consistent BA code is reported / the total BA codes reported)
+      is less than 80%. This is meant to find plants where there are multiple years of
+      an older BA code before it switches - assuming a BA code being reported for
+      multiple years is not a reporting error. TODO: make this one more explicit.
+
+    Args:
+        plants: table of annual plant attributes, including ``balancing_authority_code_eia``
+    """
+
+    def log_current_ba_code_nulls(plants: pd.DataFrame, method_str: str) -> None:
+        """Internal function to log progress on fillin in BA codes.
+
+        Args:
+            plants: the current plants table to check
+            method_str: A description of the method employed. This will be inserted into
+                the log.
+        """
+        currently_null_len = len(plants[plants.balancing_authority_code_eia.isnull()])
+        logger.info(
+            f"Filled BA codes where {method_str}. "
+            f"Currently {currently_null_len/len(plants):.1%} of records ({currently_null_len}) with no BA codes"
+        )
+
+    plants = make_backfilled_ba_code_column(plants, by_cols=["plant_id_eia"]).pipe(
+        make_consistent_ba_code_column
+    )
+    log_current_ba_code_nulls(plants, method_str="no treatment had been applied")
+    # when the backfilled code and the static code are the same, use either result
+    plants.loc[
+        (
+            plants.balancing_authority_code_eia_bfilled
+            == plants.balancing_authority_code_eia_consistent
+        )
+        & (plants.balancing_authority_code_eia.isnull()),
+        "balancing_authority_code_eia",
+    ] = plants.balancing_authority_code_eia_consistent
+
+    log_current_ba_code_nulls(
+        plants, method_str="backfilling and consistent value is the same"
+    )
+    # we found a pattern of 2013 plants reporting BA codes of PACE when the
+    # consistent option was PACW and the states the plants were located in
+    # are PACW states
+    # use PACW for those records
+    plants.loc[
+        (plants.balancing_authority_code_eia.isnull())
+        & (plants.balancing_authority_code_eia_consistent == "PACW")
+        & (plants.state.isin(["OR", "CA"])),
+        "balancing_authority_code_eia",
+    ] = "PACW"
+    log_current_ba_code_nulls(
+        plants, method_str="consistent code is PACW and state is OR or CA"
+    )
+    # found the opposite! Plants in UT labeled as PACW instead of PACE
+    plants.loc[
+        (plants.balancing_authority_code_eia.isnull())
+        & (plants.balancing_authority_code_eia_consistent == "PACE")
+        & (plants.state.isin(["UT"])),
+        "balancing_authority_code_eia",
+    ] = "PACE"
```

Comment:
Would it be worth looking more broadly at what states are typically in the East and West territories and adding them to the lists of states that can be corrected this way just in case? I feel like I saw these territories flipflopping around and impacting NV and eastern CA too.

Reply:
these have been moved.